File: packages/proposal-sign-extension-ops/src/polyfills/i32_extend8_s.c
#include <webassembly.h>
#include "stdint.h"

export int32_t i32_extend8_s(int32_t x) {
  return (x & 0x00000080) ? (x | 0xffffff80) : (x & 0x0000007f);
}

File: packages/proposal-sign-extension-ops/src/polyfills/i32_extend8_s.ts
/**
 * Polyfill for i32_extend8_s instruction
 */
export function i32_extend8_s(x: i32): i32 {
	return (x & 0x00000080) ? (x | 0xffffff80) : (x & 0x0000007f);
}
